Remove `--preview` as a required argument for `ruff server`

snowsignal opened the pull request:
## Summary

`ruff server` has reached a point of stabilization, and `--preview` is no longer required as a flag.

`--preview` is still supported as a flag, since future features may be need to gated behind it initially.

## Test Plan

A simple way to test this is to run `ruff server` from the command line. No error about a missing `--preview` argument should be reported.

MichaReiser reviewed (approved):
I'm okay merging this but there are at least two follow-ups that are necessary

* Don't set the `--preview` flag in `ruff-vscode` if the `server` version is `>= 0.5`. We should do this pre-release
* I would expect that `ruff server --preview` works the same as `ruff check --preview` and `ruff format --preview` and enables all `--preview` features (linter and formatter). 

I would expect that running `ruff server --preview` would enable all preview features similar to setting `preview = true` in the settings.

snowsignal commented:
@MichaReiser Thanks for the review! I'll look into addressing the points you brought up.

> I would expect that running `ruff server --preview` would enable all preview features similar to setting `preview = true` in the settings.

We also have server settings, `lint.preview`, and `format.preview`, that control this behavior. We should make it clear that this would override those settings.
